app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -317,7 +335,9 @@ public void onTimelineChanged(Timeline timeline, Object manifest) {
 
         @Override
         public void onTracksChanged(TrackGroupArray trackGroups, TrackSelectionArray trackSelections) {
-            // Do nothing
+            if (sharedPref.getBoolean("settings_clear_backbuffer_on_track_change", true)) {
```

Comment:
I think there is a misunderstanding here. `onTracksChanged` is probably not something you think it is. I'm talking about the case when station broadcasts live metadata - see `IcyDataSource.readMetaData` and of course about the case when station is changed. I think neither of them can trigger `onTracksChanged`.

Reply:
Indeed, I misread `onTracksChanged` as `onTrackChanged`.

Anyway, `IcyDataSource.readMetaData` should trigger  a new `onTrackChanged` listener or maybe rather  `onTitleChanged` (in order not to promise too much) then, if the stream title has changed, or what do you think?

Reply:
`ExoPlayerWrapper` already has `onDataSourceStreamLiveInfo` you can check if something changed there.

Reply:
I know, but just from updated and even changes track / title you can only make I wild guess if a track change has happened, so that I would at least put the detection and heuristics in a separate function.

Radio stations seem to alternately write their phone number, web address, or what ever into the metadata.

Reply:
Yes, we cannot know 100% sure that track changed, we can only know when the station changes a song name however most stations do that and in case some don't - nothing bad happens.

> so that I would at least put the detection and heuristics in a separate function.

Ok